src/execute/oas3/style-serializer.js

```diff
@@ -159,29 +161,24 @@ function encodeObject({ key, value, style, explode, escape }) {
 }
 
 function encodePrimitive({ key, value, style, escape }) {
-  const valueEncoder = (str) =>
-    encodeDisallowedCharacters(str, {
-      escape,
-    });
-
   if (style === 'simple') {
-    return valueEncoder(value);
+    return valueEncoder(value, escape);
   }
 
   if (style === 'label') {
-    return `.${valueEncoder(value)}`;
+    return `.${valueEncoder(value, escape)}`;
   }
 
   if (style === 'matrix') {
-    return `;${key}=${valueEncoder(value)}`;
+    return `;${key}=${valueEncoder(value, escape)}`;
   }
 
   if (style === 'form') {
-    return valueEncoder(value);
+    return valueEncoder(value, escape);
   }
 
   if (style === 'deepObject') {
-    return valueEncoder(value, {}, true);
+    return valueEncoder(value, escape);
```

Comment:
While refactoring `valueEncoder` as a separate function, I noticed that what was there before 
```
return valueEncoder(value, {}, true);
``` 
shouldn't actually pass these additional parameters to the function, as there's only one defined. It looks to me like `{}` was supposed to be a falsy `encode` parameter and `true` was to be the `parse` parameter in `encodeDisallowedCharacters`.

I found the PR for this: https://github.com/swagger-api/swagger-js/pull/1235/files and it looks to me like these values were never actually used.

Looking further into this.